src/lib/provable/crypto/encryption.ts

```diff
@@ -68,3 +73,108 @@ function decrypt(
 
   return message;
 }
+
+/**
+ * Decrypts a {@link CipherText} using a {@link PrivateKey}.
+ */
+function decryptV2(
+  {
+    publicKey,
+    cipherText,
+    messageLength,
+  }: CipherText & { messageLength: number },
+  privateKey: PrivateKey
+) {
+  // key exchange
+  const sharedSecret = publicKey.scale(privateKey.s);
+  const sponge = new Poseidon.Sponge();
+  sponge.absorb(sharedSecret.x);
+  const authenticationTag = cipherText.pop();
+
+  // decryption
+  const message = [];
+  for (let i = 0; i < cipherText.length; i++) {
+    // absorb frame tag
+    if (i === cipherText.length - 1) sponge.absorb(Field(1));
+    else sponge.absorb(Field(0));
+
+    const keyStream = sponge.squeeze();
+    const messageChunk = cipherText[i].sub(keyStream);
+
+    // convert to bytes
+    const byteMessage = wordToBytes(messageChunk, 32);
```

Comment:
wait shouldn't this be `wordToBytes(messageChunk, 31)`? I'm confused

Reply:
upps, thats right!